pkg/restore/split_test.go

```diff
@@ -316,6 +319,36 @@ FindRegion:
 	return true
 }
 
+func (s *testRestoreUtilSuite) TestGetSplitKeys(c *C) {
+	rewriteRules := &restore.RewriteRules{}
+	regions := []*restore.RegionInfo{
+		{
+			// last region's endkey is empty.
+			Region: &metapb.Region{
+				Id:       1,
+				StartKey: codec.EncodeBytes([]byte{}, []byte("t1")),
+				EndKey:   nil,
+			},
+		},
+	}
+	r := ranger.FullIntRange(false)
+	tableID := int64(1)
+	kvRanges := distsql.TableRangesToKVRanges(tableID, r, nil)
+	ranges := []rtree.Range{
+		{
+			StartKey: kvRanges[0].StartKey,
+			// this endkey is HighInclude, so it will generate one more zero byte.
+			EndKey:   kvRanges[0].EndKey,
+		},
+	}
+	keyMap := restore.GetSplitKeys(rewriteRules, ranges, regions)
+	c.Assert(len(keyMap[1]), Equals, 1)
+	// this key should truncated already, then it's as same as HighExclude endkey.
```

Comment:
Does truncateRowKey compatible with clustered index?

Reply:
4.0 doesn't have feature clustered index? so it's fine.